src/opm/output/eclipse/FileService/EclOutput.cpp

```diff
@@ -400,3 +449,24 @@ void EclOutput::writeFormattedCharArray(const std::vector<std::string>& data)
         ofileH  << std::endl;
     }
 }
+
+void EclOutput::writeFormattedCharArray(const std::vector<Opm::RestartIO::Helpers::CharArrayNullTerm<8>>& data)
+{
+    const auto sizeData = EIOD::block_size_data_formatted(EIOD::CHAR);
+
+    const int nColumns = std::get<1>(sizeData);
+
+    const auto size = data.size();
+
+    for (auto i = 0*size; i < size; ++i) {
+        ofileH << " '" << data[i].c_str() << '\'';
```

Comment:
Is the `char8NullTerminated ...` type guaranteed to be space padded to length 8? Otherwise that must be done here.

Reply:
> Is the `char8NullTerminated` ... type guaranteed to be space padded to length 8?

Yes. That's its only reason the class exists.